src/adapters/places.ts

```diff
@@ -0,0 +1,85 @@
+import { AppComponents, PlaceNotFoundError } from '../types'
+import { IPlacesComponent, PlaceAttributes, PlaceResponse } from '../types/places.type'
+
+export async function createPlacesComponent(
+  components: Pick<AppComponents, 'config' | 'cachedFetch' | 'logs'>
+): Promise<IPlacesComponent> {
+  const { config, cachedFetch, logs } = components
+
+  const logger = logs.getLogger('places-component')
+
+  const [placesApiUrl] = await Promise.all([config.requireString('PLACES_API_URL')])
+
+  const fetchFromCache = cachedFetch.cache<PlaceResponse>()
+  async function getPlaceByParcel(parcel: string): Promise<PlaceAttributes> {
+    cachedFetch.cache
+
+    const response = await fetchFromCache.fetch(`${placesApiUrl}/places?positions=${parcel}`)
+
+    if (!response?.ok) {
+      logger.info(`Error getting place information: ${parcel}`)
+      throw new Error(`Error getting place information: ${parcel}`)
+    }
```

Comment:
Can this happen? I think the cache fetcher is already handling this case.

Reply:
good catch